### ISSUE TITLE
keysnail.service: Failed with result 'exit-code' on Ubuntu 22.04.1

### ISSUE THREAD

**Issue by nabelk** [rbreaves/kinto#765]
Everything went smooth until after the setup wizard, remapping cannot be done. There's a looping (screenshot below)

**Distro:** Ubuntu 22.04.1 LTS
**DE:** Gnome

![Screenshot from 2022-11-03 04-21-35](https://user-images.githubusercontent.com/107866604/199597804-f8226980-dfad-4095-84a6-7f22cf082bd9.png)

**Comment by yanksyoon** [comment]
Hi, I was able to solve the problem,
The problem on my side was that python3/python, pip, pip3 were not pointing to the same modules due to anaconda installation.
(`which python3` and `sudo which python3` would differ).
So the first thing I did was deactivate conda environment(`conda deactivate`).

After you deactivate conda and install kinto successfully (with GUI popup), you will see it trigger the journalctl as shown in your creenshot.

I then fixed the problem by installing tkinter using the same shell via the command `sudo apt-get install python3-tk`

then i ran the original command triggered by the `kinto.sh` (just press up until you see the original command).

Everything works like magic now :)

Hope this helps!

**Comment by nabelk** [comment]
@yanksyoon Still not working, I already check there's no anaconda install in my system.
And I did try to install tkinter but after that, my keyboard stopped working. So I have to stop `kinto.sh` process.

Are you using Ubuntu 22.04 LTS (wayland)?

I changed to xorg on Ubuntu 22.04 to make the kinto installation but still can't run on it neither on wayland.

**Comment by RedBearAK** [comment]
@nabelk 

The keymapper application (xkeysnail or keyszer) uses Xlib to get the window class and name from X11/Xorg environments. There's no sign yet of a way to get this information about windows in Wayland. The installer script tries to stop Kinto from being installed in Wayland, but if you do happen to install under X11/Xorg and then log back into Wayland you may notice the only thing that really works is the swapping of the modifier keys. None of the individual app or terminal remaps of specific shortcuts will work correctly under Wayland. 

In short, Kinto is not compatible with Wayland yet.

**Comment by RedBearAK** [comment]
@nabelk

I replicated the issue with a fresh install of Ubuntu from the 22.04.1 updated ISO, in a Boxes VM. It was a minimal install, no updates applied during install, no Anaconda or anything installed before trying to install Kinto. 

There is loss of keyboard input, but only in the terminal window where you run the installer, and this persists even after using the Kinto GUI app to stop Kinto. Closing that terminal window and opening a new one will get the keyboard input back. Other app windows don't seem to be affected, at least after stopping the Kinto service. 

After that, this command seems to completely solve the problem: 

```
sudo apt install python3-tk
```

@rbreaves 

Looks like the installer script needs to make sure `python3-tk` is installed on Ubuntu 22.04.1 and later.

**Comment by rbreaves** [comment]
Closing, has been added for all Ubuntu versions now. Thanks @RedBearAK and to the others figuring this out.

**Comment by HuangJiaLian** [comment]
> Hi, I was able to solve the problem, The problem on my side was that python3/python, pip, pip3 were not pointing to the same modules due to anaconda installation. (`which python3` and `sudo which python3` would differ). So the first thing I did was deactivate conda environment(`conda deactivate`).
> 
> After you deactivate conda and install kinto successfully (with GUI popup), you will see it trigger the journalctl as shown in your creenshot.
> 
> I then fixed the problem by installing tkinter using the same shell via the command `sudo apt-get install python3-tk`
> 
> then i ran the original command triggered by the `kinto.sh` (just press up until you see the original command).
> 
> Everything works like magic now :)
> 
> Hope this helps!

My problem seems happened after installing Anaconda too. Here is how I solved it in a slightly different way. 

```bash
sudo apt install python3-evdev
```

Since the error message said cannot find evdev. 

![image](https://github.com/rbreaves/kinto/assets/17427791/40f6a872-690c-4b6b-a507-11ff2731b338)
